calledudeBot/Bots/IrcClient.cs

```diff
@@ -97,11 +97,40 @@ protected async Task Login()
                 }
                 if (result == 001)
                 {
-                    await WriteLine($"JOIN {channelName}");
-                    if (OnReady != null)
-                        await OnReady.Invoke();
+                    if(channelName != null)
+                        await WriteLine($"JOIN {channelName}");
 
-                    if (!TestRun) TryLog($"Connected to {Name}-IRC.");
+                    if (Ready != null)
+                        await Ready.Invoke();
+
+                    if (!TestRun)
+                    {
+                        TryLog($"Connected to {Name}-IRC.");
+                    }
+                }
+            }
+        }
+
+        protected async Task Listen()
+        {
+            while (true)
+            {
+                var buffer = await input.ReadLineAsync();
+                var b = buffer.Split(' ');
```

Comment:
Will this be fine if you don't do `StringSplitOptions.RemoveEmptyEntries`? Or do your parsers account for that

Reply:
I mean, IRC messages are structured in a certain way, and that can't be changed by anyone else than the IRC server itself, so it'll be fine :p